src/transformers/trainer.py

```diff
@@ -225,9 +226,16 @@ def __init__(
                 ),
                 FutureWarning,
             )
+        # Some warning for the user related to the usage of datasets that do not implement __len__
+        if args.max_steps > 0:
+            logger.warning("max_steps is given, it will override any value given in num_train_epochs")
+        if not isinstance(train_dataset, collections.abc.Sized) and args.max_steps == 0:
```

Comment:
You should check there is a `train_dataset` by adding `train_dataset is not None` in your test (the Trainer could be used just for eval/predictions).

Reply:
Done.